### DISCUSSION TITLE
Lazer defaults to second monitor instead of primary.

### DISCUSSION THREAD

**Discussion by TheFantasticLoki** [ppy/osu#17039] [Q&A] [answered]
So this is a wierd issue that only appeared after disabling g-sync but enabling g-sync again does not resolve this issue. I have also checked both nvidia control panel and windows 10 display settings, both report the correct monitor numbering and positioning. (Second monitor on the left Primary on the right)

I am attaching all the logs from lazer tho I kinda doubt anything will be reported in them. I'd like advice on how I can obtain more diagnostic reports and information for you to examine.
[performance.log](https://github.com/ppy/osu/files/8166646/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8166647/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8166648/updater.log)
[database.log](https://github.com/ppy/osu/files/8166649/database.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/8166650/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/8166651/network.log)

**Comment by peppy** [comment, marked as answer]
I'm guessing you are running the game in full screen mode?

Can you try switching to windowed, dragging to your primary display and then enabling full screen again to see if this resolves? We eventually need a monitor selection in the settings, but until that is done, I believe following this procedure should save the correct monitor assignment again.

**Reply by TheFantasticLoki** [reply]
Thanks for the quick reply once again this did resolve the issue marking as answered. Have a good day peppy.

**Reply by peppy** [reply]
Will be better solved via #17047.

**Reply by TheFantasticLoki** [reply]
Quick update: I was able to recreate the issue and it stems from the way displayport functions. If you Leave lazer open say in the background overnight and turn off your monitors, The primary displayport monitor will register to the computer as unplugged. Whereas the second monitor plugged in via HDMI will always register as being plugged in. If lazer is still open it will switch to the second monitor as it now registers this as the only attached display. This is why it starts opening on the secondary monitor based on my testing and understanding.
Should be fixed by #17047 but I wanted to explain the original cause.
